Source/WebKit/WebProcess/GPU/graphics/RemoteRenderingBackendProxy.cpp

```diff
@@ -83,41 +86,44 @@ RemoteRenderingBackendProxy::~RemoteRenderingBackendProxy()
     if (!m_streamConnection)
         return;
 
-    ensureOnMainRunLoop([ident = renderingBackendIdentifier()]() {
-        WebProcess::singleton().ensureGPUProcessConnection().connection().send(Messages::GPUConnectionToWebProcess::ReleaseRenderingBackend(ident), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
+    ensureOnMainRunLoop([identifier = m_identifier, weakGPUProcessConnection = WTFMove(m_gpuProcessConnection)]() {
+        RefPtr gpuProcessConnection = weakGPUProcessConnection.get();
+        if (!gpuProcessConnection)
+            return;
+        gpuProcessConnection->releaseRenderingBackend(identifier);
     });
     m_remoteResourceCacheProxy.clear();
     disconnectGPUProcess();
 }
 
 void RemoteRenderingBackendProxy::ensureGPUProcessConnection()
 {
-    if (!m_streamConnection) {
-        static constexpr auto connectionBufferSizeLog2 = 21;
-        auto connectionPair = IPC::StreamClientConnection::create(connectionBufferSizeLog2);
-        if (!connectionPair)
-            CRASH();
-        auto [streamConnection, serverHandle] = WTFMove(*connectionPair);
-        m_streamConnection = WTFMove(streamConnection);
-        // RemoteRenderingBackendProxy behaves as the dispatcher for the connection to obtain isolated state for its
-        // connection. This prevents waits on RemoteRenderingBackendProxy to process messages from other connections.
-        m_streamConnection->open(*this, *this);
-
-        callOnMainRunLoopAndWait([this, serverHandle = WTFMove(serverHandle)]() mutable {
-            auto& gpuProcessConnection = WebProcess::singleton().ensureGPUProcessConnection();
-            m_connection = &gpuProcessConnection.connection();
-            m_sharedResourceCache = gpuProcessConnection.sharedResourceCache();
-            m_connection->send(Messages::GPUConnectionToWebProcess::CreateRenderingBackend(m_parameters, WTFMove(serverHandle)), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
-        });
-    }
+    if (m_streamConnection)
+        return;
+    static constexpr auto connectionBufferSizeLog2 = 21;
+    auto connectionPair = IPC::StreamClientConnection::create(connectionBufferSizeLog2);
+    if (!connectionPair)
+        CRASH();
+    auto [streamConnection, serverHandle] = WTFMove(*connectionPair);
+    m_streamConnection = WTFMove(streamConnection);
+    // RemoteRenderingBackendProxy behaves as the dispatcher for the connection to obtain isolated state for its
+    // connection. This prevents waits on RemoteRenderingBackendProxy to process messages from other connections.
+    m_streamConnection->open(*this, *this);
+
+    callOnMainRunLoopAndWait([&]() {
+        auto& gpuProcessConnection = WebProcess::singleton().ensureGPUProcessConnection();
+        gpuProcessConnection.createRenderingBackend(m_identifier, WTFMove(serverHandle));
```

Comment:
My quick reading of the code says this is probably safe since there is a wait, and therefore the object is still in scope, but the compiler doesn't like it:
```
RemoteRenderingBackendProxy.cpp:115:75: error: reference to local binding 'serverHandle' declared in enclosing function 'WebKit::RemoteRenderingBackendProxy::ensureGPUProcessConnection'
```